plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSharedGlueMetastoreWithTableRedirections.java

```diff
@@ -108,4 +119,58 @@ protected String getExpectedDeltaLakeCreateSchema(String catalogName)
                 ")";
         return format(expectedDeltaLakeCreateSchema, catalogName, schema, dataDirectory.toUri());
     }
+
+    @Test
+    public void testUnsupportedHiveTypeRedirect()
+    {
+        String tableName = "unsupported_types";
+        String location = dataDirectory.toUri() + "/" + tableName;
+        StorageDescriptor sd = StorageDescriptor.builder()
+                .columns(Column.builder()
+                        .name("last_hour_load")
+                        .type("timestamp_ntz")
+                        .build())
+                .location(location)
+                .inputFormat("org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe")
+                .outputFormat("org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat")
+                .serdeInfo(SerDeInfo.builder()
+                        .serializationLibrary("org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe")
+                        .parameters(Map.of(
+                                "serialization.format", "1",
+                                "path", location))
+                        .build())
+                .build();
+        TableInput tableInput = TableInput.builder()
+                .name(tableName)
+                .storageDescriptor(sd)
+                .parameters(Map.of(
+                        "EXTERNAL", "TRUE",
+                        "delta.minReaderVersion", "3",
+                        "delta.maxReaderVersion", "7",
+                        "spark.sql.sources.schema", "{\"type\":\"struct\",\"fields\":[{\"name\":\"last_hour_load\",\"type\":\"timestamp_ntz\",\"nullable\":true,\"metadata\":{}}]}",
+                        "spark.sql.partitionProvider", "catalog",
+                        "delta.feature.timestampNtz", "supported",
+                        "spark.sql.sources.provider", "delta",
+                        "delta.lastCommitTimestamp", "1720682605000",
+                        "spark.sql.create.version", "3.4.1",
+                        "delta.lastUpdateVersion", "0"))
+                .tableType("EXTERNAL_TABLE")
+                .build();
+
+        CreateTableRequest createTableRequest = CreateTableRequest.builder()
+                .databaseName(schema)
+                .tableInput(tableInput)
+                .build();
+        try (GlueClient glueClient = GlueClient.create()) {
+            glueClient.createTable(createTableRequest);
+
+            // The table is incomplete, so it is expected to fail after reading the metadata from Glue
+            assertQueryFails("SHOW CREATE TABLE hive_with_redirections." + schema + "." + tableName, "Metadata not found in transaction log for " + schema + "." + tableName);
```

Comment:
Can you try to prepare table, so that query succeeds?

Reply:
Done!